### PR TITLE
Make an attempt at organizing hdwallet provider files

### PR DESCRIPTION
**Pull request by eggplantzzz**

### CODE REVIEW

File: packages/hdwallet-provider/src/LegacyConstructor.ts
Comment (gnidan) [inline packages/hdwallet-provider/src/LegacyConstructor.ts]: since we're making this a module of its own, we don't need the namespace - we can just have all these be top-level exports, and then do `import * as LegacyConstructor from "./LegacyConstructor"` on the import side.

File: packages/hdwallet-provider/src/LegacyConstructor.ts
Comment (gnidan) [inline packages/hdwallet-provider/src/LegacyConstructor.ts]: (as per the other comment, this would be `import * as NewConstructor`)

**Comment by eggplantzzz** [comment]
https://github.com/trufflesuite/truffle/pull/3209

File: packages/hdwallet-provider/src/LegacyConstructor.ts
Comment (eggplantzzz) [reply packages/hdwallet-provider/src/LegacyConstructor.ts]: That makes a lot of sense. This will make me feel better about naming these files something like `constructor.ts` and `legacyConstructor.ts`.

File: packages/hdwallet-provider/src/types.ts
Comment (gnidan) [inline packages/hdwallet-provider/src/types.ts]: This being here makes it a circular import (since `types` imports `legacyConstructor` imports `types`, e.g.)

We should probably move this type somewhere else to avoid this.

**Comment by richardpringle** [comment]
LGTM
@gnidan, let us know when you've had another chance to take a look and I can merge

**Comment by richardpringle** [comment]
@eggplantzzz @gnidan I invited you to be maintainers: https://github.com/richardpringle/truffle/invitations
Once the branch is good, you shouldn't have to wait for me to merge

**Review by gnidan — APPROVED** [review]
LGTM!

**Review by richardpringle — APPROVED** [review]
Just need to take this out of DRAFT to merge

**Comment by gnidan** [comment]
I'll do it :)

**Pull request merged**
